src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -450,6 +449,229 @@ static void propagate_padding_to_opt_out_users(program_node& node, cldnn::paddin
     }
 }
 
+bool crop_in_place_optimization::match(crop_node& node) {
+    auto pred_param = node.get_dependency(0).get_kernel_impl_params();
+    auto pred_layout = pred_param->get_output_layout();
+    return (match(node, *node.get_kernel_impl_params(), pred_layout));
+}
+
+bool crop_in_place_optimization::match(const program_node& node,
+                                       kernel_impl_params& crop_params,
+                                       layout& input_layout,
+                                       bool is_runtime) {
+    if (!node.is_valid_output_layout())
+        return false;
+    // if the node is marked as network output, prevent optimizations which would affect a form of its output,
+    // unless debug flag is set
+    if (node.is_output() || crop_params.fused_desc.size() > 0 || node.is_in_shape_of_subgraph())
+        return false;
+
+    const auto& crop_layout = crop_params.get_output_layout();
+    for (auto user : node.get_users()) {
+        // do not optimize when next node is concatenation which is not output
+        if (user->is_type<concatenation>() && !user->is_output())
+            return false;
+        if (user->is_type<loop>() || user->is_type<non_max_suppression>())
+            return false;
+        // If the input tensor of convolution includes dynamic padding, there is an issue
+        // where the total size of tensor is not properly calculated and becomes 0
+        // It causes issue for internal buffer allocation during runtime
+        // TODO: Need to allow optimization for gemm user
+        if (node.is_dynamic() && (user->is_type<convolution>() || user->is_type<gemm>()))
+            return false;
+        if (user->is_type<reshape>()) {
+            // runtime buffer fusing is only handled when there is only one reshape user
+            if (node.is_dynamic() && node.get_users().size() != 1)
+                return false;
+            auto& reshape_node = user->as<reshape>();
+            if (!reshape_node.is_runtime_propagatable_padding() && can_reshape_be_optimized(reshape_node))
+                return false;
```

Comment:
For static shape, it was always returning false for can_reshape_be_optimized is true. I think we shouldnt change static shape's behavior if it is not fully validated.

Reply:
done